community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/CommunityLockManger.java

```diff
@@ -25,10 +25,18 @@
 public class CommunityLockManger extends LifecycleAdapter implements Locks
 {
     private final LockManagerImpl manager = new LockManagerImpl( new RagManager() );
+    private volatile boolean closed;
 
     @Override
     public Client newClient()
     {
+        // We check this volatile closed flag here, which may seem like a contention overhead, but as the time
+        // of writing we apply pooling of transactions and in extension pooling of lock clients,
+        // so this method is called very rarely.
+        if ( closed )
```

Comment:
why do we need this flag, lock managers already have `LockClientStateHolder` that should be able to track state and throws exceptions when it will be used if closed?